bofire/strategies/doe_strategy.py

```diff
@@ -0,0 +1,52 @@
+import pandas as pd
+from pydantic.types import PositiveInt
+
+import bofire.data_models.strategies.api as data_models
+from bofire.strategies.doe.design import find_local_max_ipopt
+from bofire.strategies.strategy import Strategy
+
+
+class DoEStrategy(Strategy):
+    """Strategy for design of experiments. This strategy is used to generate a set of
+    experiments for a given domain.
+    The experiments are generated via minimization of the D-optimality criterion.
+
+    """
+
+    def __init__(
+        self,
+        data_model: data_models.DoEStrategy,
+        **kwargs,
+    ):
+        super().__init__(data_model=data_model, **kwargs)
+        self.formula = data_model.formula
+        self.domain = data_model.domain
+
+    def _ask(self, candidate_count: PositiveInt) -> pd.DataFrame:
+        _fixed_experiments = self.candidates
+        _domain = self.domain
+
+        if _fixed_experiments is not None:
```

Comment:
Short question: why do you do this with all the underscores, why not just:

`if self.candidates is None:`?